### PR TITLE
Add AAD app for external connections

### PR DESCRIPTION
**Pull request by tanya-borisova**
## What is being addressed

Adds a new AAD app, `flowehr-external-${suffix}`, that is used to connect FlowEHR with downstream systems.

The specific way we want to use this app is to assign it roles to resources in TRE workspaces, so that Databricks Pipelines can use credentials from this one single app, rather than having to manipulate credential per each workspace and resource.

## How is this addressed

- Added a new AAD app, flowehr_external_connection, added corresponding service principal and a secret
- Create secrets in Databricks for this app
- Push the secrets to keyvault
- Refactor all AAD apps in Transform to live in auth.tf file

### CODE REVIEW

**Review by damoodamoo — APPROVED** [review]
like it, refactoring is neater too 👍

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/6071183535 (with refid `0667af66`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1704860927) from @damoodamoo)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

`/test-destroy-env` is not recognised as a valid command.

You can use the following commands:
&nbsp;&nbsp;&nbsp;&nbsp;/test - deploy any modified modules and destroy
&nbsp;&nbsp;&nbsp;&nbsp;/test-all - deploy everything (ignoring path filters) and destroy
&nbsp;&nbsp;&nbsp;&nbsp;/test-force-approve - force approval of the PR tests (i.e. skip the deployment checks)
&nbsp;&nbsp;&nbsp;&nbsp;/destroy - delete the PR environment (e.g. to enable testing a deployment from a clean start after previous tests)
&nbsp;&nbsp;&nbsp;&nbsp;/destroy-no-terraform - delete the PR environment without terraform
&nbsp;&nbsp;&nbsp;&nbsp;/help - show this help

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1704934599) from @tanya-borisova)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

Destroying environment (with refid 0667af66)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1704935916) from @tanya-borisova)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/6072118095 (with refid `0667af66`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1704998877) from @tanya-borisova)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

Destroying environment no terraform (with refid 0667af66).	:warning: This will leave orphaned resources

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1705034480) from @tanya-borisova)

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/6073232663 (with refid `0667af66`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1705163415) from @tanya-borisova)

**Comment by tanya-borisova** [comment]
/test-force-approve

I didn't manage to run tests in the environment unfortunately, but I can confirm that it works in the freshly deployed environment I made

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:white_check_mark: Marking tests as complete (for commit 7055d041032df7bb60af51d64cbf4b042d147572)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/324#issuecomment-1705422018) from @tanya-borisova)

File: infrastructure/transform/secrets.tf [hunk 1]
```diff
@@ -70,6 +70,18 @@ resource "databricks_secret" "flowehr_databricks_sql_database" {
   string_value = azurerm_mssql_database.feature_database.name
   scope        = databricks_secret_scope.secrets.id
 }
+# FlowEHR external app secrets
```
Comment (jjgriff93) [inline infrastructure/transform/secrets.tf:73]:
```suggestion

# FlowEHR external app secrets
```

**Review by jjgriff93 — APPROVED** [review]
Looks great!

**Pull request merged**